src/Discord/Internal/Types/ApplicationCommands.hs

```diff
@@ -696,3 +797,74 @@ instance ToJSON ApplicationCommandPermissions where
         "type" .== applicationCommandPermissionsType,
         "permission" .== applicationCommandPermissionsPermission
       ]
+
+data Locale
+  = Da | De | EnGB | EnUS | EsES | Fr | Hr | It | Lt | Hu | Nl | No | Pl | PtBR
+  | Ro | Fi | SvSE | Vi | Tr | Cs | El | Bg | Ru | Uk | Hi | Th | ZhCN | Ja
+  | ZhTW | Ko
+  deriving (Show, Read, Eq, Ord)
+
+type TextTranslation = (Locale, T.Text)
+
+newtype LocalizedText = LocalizedText [TextTranslation] deriving (Show, Read, Eq, Ord)
+
+instance ToJSON LocalizedText where
+  toJSON (LocalizedText lt) = object . map (\(l,t) -> (localeString l, toJSON t)) $ lt
+    where
+      localeString l = case l of
+        Da -> "da"
+        De -> "de"
```

Comment:
I really think it would be worth somehow having a bidirectional map for this stuff so that the two locale lists don't get out of sync. alternatively, we just use Texts to represent a locale and don't need us to do any translations

Reply:
Maybe just going with text might be smarter actually yeah